src/sentry/sentry_metrics/querying/data/execution.py

```diff
@@ -370,6 +447,11 @@ def _execute(
                         modified_start, modified_end
                     )
 
+            # If, after merging the query with the blocked projects, the query becomes empty, we will return an empty
+            # result.
+            if executable_query.is_empty():
```

Comment:
We do this check after the time interval adjustment so that we return the right interval data in the response.